pliers/extractors/base.py

```diff
@@ -120,9 +120,15 @@ def to_df(self, timing=True, metadata=False, format='wide',
                             for i in range(data.shape[1])]
             df = pd.DataFrame(data, columns=features)
 
-        onsets = np.nan if self.onset is None else self.onset
-        durations = np.nan if self.duration is None else self.duration
-        orders = np.nan if self.order is None else self.order
+        if hasattr(self, '_onsets'):
+            onsets = np.array(self._onsets)
+            onsets += 0.0 if self.onset is None else self.onset
+        else:
+            onsets = np.nan if self.onset is None else self.onset
+        durations = getattr(self, '_durations', self.duration)
+        durations = np.nan if durations is None else durations
+        orders = getattr(self, '_orders', self.order)
+        orders = np.nan if orders is None else orders
```

Comment:
Manually checks to see if the Extractor has set new custom timing attributes (prepended with _) during a call to `_to_df`